components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -89,7 +91,11 @@ public Response issueAccessToken(@Context HttpServletRequest request, String pay
 
         Map<String, List<String>> paramMap;
         try {
-            startSuperTenantFlow();
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                startTenantFlow();
```

Comment:
I don't think we need to start the tenant flow if the Tenant Qualified Urls are enabled. Please add the following logic and test the flow as if the tenant-qualified URLs are enabled it resolves the tenant path.
if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
 startSuperTenantFlow();
}

Reply:
Fixed